src/main/java/uk/gov/companieshouse/itemhandler/kafka/OrdersKafkaConsumer.java

```diff
@@ -1,173 +1,261 @@
     private static final String ORDER_RECEIVED_GROUP = APPLICATION_NAMESPACE + "-" + ORDER_RECEIVED_TOPIC;
     private static final String ORDER_RECEIVED_GROUP_RETRY = APPLICATION_NAMESPACE + "-" + ORDER_RECEIVED_TOPIC_RETRY;
     private static final String ORDER_RECEIVED_GROUP_ERROR = APPLICATION_NAMESPACE + "-" + ORDER_RECEIVED_TOPIC_ERROR;
-    private static final int MAX_RETRY_ATTEMPTS = 3;
+    private static long errorRecoveryOffset = 0l;
 
-    private CHKafkaResilientConsumerGroup chKafkaConsumerGroupMain;
-    private CHKafkaResilientConsumerGroup chKafkaConsumerGroupRetry;
     @Value("${spring.kafka.consumer.bootstrap-servers}")
-    private String brokerAddresses;
+    private String bootstrapServers;
+    @Value("${uk.gov.companieshouse.item-handler.error-consumer}")
+    private boolean errorConsumerEnabled;
     private final SerializerFactory serializerFactory;
+    private final OrdersKafkaProducer kafkaProducer;
+    private final KafkaListenerEndpointRegistry registry;
 
-    public OrdersKafkaConsumer(SerializerFactory serializerFactory) {
+    public OrdersKafkaConsumer(SerializerFactory serializerFactory,
+                               OrdersKafkaProducer kafkaProducer,
+                               KafkaListenerEndpointRegistry registry) {
         this.serializerFactory = serializerFactory;
+        this.kafkaProducer = kafkaProducer;
+        this.registry = registry;
     }
 
-    @Override
-    public void afterPropertiesSet() throws Exception {
-        LOGGER.debug("Initializing kafka consumer service " + this.toString());
-
-        ProducerConfig config = getProducerConfig();
-        CHKafkaProducer chKafkaProducer = new CHKafkaProducer(config);
-
-        ConsumerConfig consumerConfig = getConsumerConfig();
-
-        consumerConfig.setGroupName(ORDER_RECEIVED_GROUP);
-        consumerConfig.setTopics(Collections.singletonList(ORDER_RECEIVED_TOPIC));
-        chKafkaConsumerGroupMain
-                = new CHKafkaResilientConsumerGroup(consumerConfig, CHConsumerType.MAIN_CONSUMER,
-                        new KafkaConsumerFactory(), chKafkaProducer);
+    /**
+     * Main listener/consumer. Calls a `retryable` method to process received message.
+     * If the `retryable` processor is unsuccessful with a `retryable` error, after maximum numbers of attempts allowed,
+     * the message is published to `-retry` topic for failover processing.
+     * @param message
+     * @throws SerializationException
+     * @throws ExecutionException
+     * @throws InterruptedException
+     */
+    @KafkaListener(id = ORDER_RECEIVED_GROUP, groupId = ORDER_RECEIVED_GROUP,
+                    topics = ORDER_RECEIVED_TOPIC,
+                    autoStartup = "#{!${uk.gov.companieshouse.item-handler.error-consumer}}")
+    public void processOrderReceived(org.springframework.messaging.Message<OrderReceived> message)
+            throws SerializationException, ExecutionException, InterruptedException {
```

Comment:
Can these methods follow the [exception handling standards](https://companieshouse.atlassian.net/wiki/spaces/DEV/pages/486736015/Exception+handling#Exceptionhandling-Howtopasserrorsupthechain) please?

Reply:
This has been updated.